Mapping/ContainerAwareEntityListenerResolver.php

```diff
@@ -2,107 +2,16 @@
 
-class ContainerAwareEntityListenerResolver implements EntityListenerServiceResolver
+/**
+ * @deprecated since 1.11 and will be removed in 2.0. Use ContainerEntityListenerResolver instead.
+ */
+class ContainerAwareEntityListenerResolver extends ContainerEntityListenerResolver
 {
-    /** @var ContainerInterface */
-    private $container;
-
-    /** @var object[] Map to store entity listener instances. */
-    private $instances = [];
-
-    /** @var string[] Map to store registered service ids */
-    private $serviceIds = [];
-
     public function __construct(ContainerInterface $container)
     {
-        $this->container = $container;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function clear($className = null)
-    {
-        if ($className === null) {
-            $this->instances = [];
-
-            return;
-        }
-
-        $className = $this->normalizeClassName($className);
-
-        if (! isset($this->instances[$className])) {
-            return;
-        }
-
-        unset($this->instances[$className]);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function register($object)
-    {
-        if (! is_object($object)) {
-            throw new InvalidArgumentException(sprintf('An object was expected, but got "%s".', gettype($object)));
-        }
-
-        $className = $this->normalizeClassName(get_class($object));
-
-        $this->instances[$className] = $object;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function registerService($className, $serviceId)
-    {
-        $this->serviceIds[$this->normalizeClassName($className)] = $serviceId;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function resolve($className)
-    {
-        $className = $this->normalizeClassName($className);
-
-        if (! isset($this->instances[$className])) {
-            if (isset($this->serviceIds[$className])) {
-                $this->instances[$className] = $this->resolveService($this->serviceIds[$className]);
-            } else {
-                $this->instances[$className] = new $className();
-            }
-        }
-
-        return $this->instances[$className];
-    }
-
-    /**
-     * @param string $serviceId
-     *
-     * @return object
-     */
-    private function resolveService($serviceId)
-    {
-        if (! $this->container->has($serviceId)) {
-            throw new RuntimeException(sprintf('There is no service named "%s"', $serviceId));
-        }
-
-        return $this->container->get($serviceId);
-    }
-
-    /**
-     * @param string $className
-     *
-     * @return string
-     */
-    private function normalizeClassName($className)
-    {
-        return trim($className, '\\');
+        @trigger_error(sprintf('The class "%s" is deprecated since 1.11 and will be removed in 2.0. Use "%s" instead.', self::class, 'Doctrine\Bundle\DoctrineBundle\Mapping\ContainerEntityListenerResolver'), E_USER_DEPRECATED);
```

Comment:
I already suspected this yesterday, but the `DebugClassLoader` indeed throws a deprecation once you have added `@deprecated` to the class above. We can drop the deprecation in the constructor 🎉